controllers/routemonitor/routemonitor_controller.go

```diff
@@ -0,0 +1,83 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routemonitor
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	monitoringv1alpha1 "github.com/openshift/route-monitor-operator/api/v1alpha1"
+)
+
+// RouteMonitorReconciler reconciles a RouteMonitor object
+type RouteMonitorReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=*,resources=services,verbs=get;list;watch;create;
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;
+// +kubebuilder:rbac:groups=monitoring.coreos.com,resources=servicemonitors,verbs=get;list;watch;create;
+// +kubebuilder:rbac:groups=monitoring.openshift.io,resources=routemonitors,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=monitoring.openshift.io,resources=routemonitors/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;list;watch
+
+func (r *RouteMonitorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	// Should happen once but cannot input in main.go
+	err := r.CreateBlackBoxResources(ctx)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
```

Comment:
as said before, I think this should always happen when the first thing is deployed, and get removed if it's not needed anymore.

Reply:
not sure where to put it if it only happens once. tried to put in main.go but wasn't executed / was too early

Reply:
@NautiluX